theatre/studio/src/uiComponents/usePresence.tsx

```diff
@@ -0,0 +1,231 @@
+import type {StudioSheetItemKey} from '@theatre/shared/utils/ids'
+import type {StrictRecord} from '@theatre/shared/utils/types'
+import React, {useMemo} from 'react'
+import {useEffect} from 'react'
+import {useLogger} from './useLogger'
+import {Box, prism, valueDerivation} from '@theatre/dataverse'
+import {Atom} from '@theatre/dataverse'
+import {useVal} from '@theatre/react'
+
+export enum Hovered {
+  /** This is the focus */
+  Hovered,
+  /** The related parent */
+  Parent,
+  /** The related child / moves with */
+  Child,
+  /** The object containing the hovered item */
+  ParentObject,
+}
+
+function createPresence(): Presence {
+  const currentPresence = new Box<StudioSheetItemKey | undefined>(undefined)
+  // TODO: move into ephemeral
+  const atom = new Atom({} as StrictRecord<StudioSheetItemKey, boolean>)
+
+  type HoveredWithImportance = {
+    importance: number
+    relationship: Hovered
+  }
+
+  // keep as part of presence creation
+  const relations = new Atom(
+    {} as StrictRecord<
+      StudioSheetItemKey,
+      StrictRecord<
+        StudioSheetItemKey,
+        StrictRecord<string, HoveredWithImportance>
+      >
+    >,
+  )
+
+  let lastRelationId = 0
+
+  return {
+    addRelationships(itemKey, relationships) {
+      const relId = String(++lastRelationId)
```

Comment:
nit: prefer just `id`, read this as `relativeId` accidentally the first time and thought "relative to what?"